cmd/crank/beta/diff/renderer/diff_renderer.go

```diff
@@ -0,0 +1,155 @@
+package renderer
+
+import (
+	"cmp"
+	"fmt"
+	"io"
+	"strings"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+
+	dt "github.com/crossplane/crossplane/cmd/crank/beta/diff/renderer/types"
+)
+
+// DiffRenderer handles rendering diffs to output.
+type DiffRenderer interface {
+	// RenderDiffs formats and outputs diffs to the provided writer
+	RenderDiffs(stdout io.Writer, diffs map[string]*dt.ResourceDiff) error
+}
+
+// DefaultDiffRenderer implements the DiffRenderer interface.
+type DefaultDiffRenderer struct {
+	logger   logging.Logger
+	diffOpts DiffOptions
+}
+
+// NewDiffRenderer creates a new DefaultDiffRenderer with the given options.
+func NewDiffRenderer(logger logging.Logger, diffOpts DiffOptions) DiffRenderer {
+	return &DefaultDiffRenderer{
+		logger:   logger,
+		diffOpts: diffOpts,
+	}
+}
+
+// SetDiffOptions updates the diff options used by the renderer.
+func (r *DefaultDiffRenderer) SetDiffOptions(options DiffOptions) {
+	r.diffOpts = options
+}
+
+func getKindName(d *dt.ResourceDiff) string {
+	// Check if the name indicates a generated name (ends with "(generated)")
+	if strings.HasSuffix(d.ResourceName, "(generated)") {
+		return fmt.Sprintf("%s/%s", d.Gvk.Kind, d.ResourceName)
+	}
+
+	// Regular case with a specific name
+	return fmt.Sprintf("%s/%s", d.Gvk.Kind, d.ResourceName)
+}
+
+// RenderDiffs formats and prints the diffs to the provided writer.
+func (r *DefaultDiffRenderer) RenderDiffs(stdout io.Writer, diffs map[string]*dt.ResourceDiff) error {
```

Comment:
Is there prior art for diffing Kubernetes resources? I'm thinking e.g. https://github.com/homeport/dyff or whatever `kubectl diff` is doing. Do we need our own bespoke implementation?